nexus/db-queries/src/db/datastore/sled.rs

```diff
@@ -564,14 +733,351 @@ impl DataStore {
                 resources.clone(),
             );
 
-            // Try to INSERT the record. If this is still a valid target, we'll
-            // use it. If it isn't a valid target, we'll shrink the set of
-            // viable sled targets and try again.
-            let rows_inserted = sled_insert_resource_query(&resource)
+            if local_storage_allocation_required.is_empty() {
+                info!(
+                    &log,
+                    "calling insert (no local storage allocation requred)"
```

Comment:
"calling insert" feels like a pretty unclear log message if you are not already intimately familiar with this code...